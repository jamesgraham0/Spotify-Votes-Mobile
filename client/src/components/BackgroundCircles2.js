import { StyleSheet, View } from 'react-native';
import COLORS from '../utils/constants';

const DarkBackgroundCircles = () => {
    return (
        <View style={styles.container}>
            <View style={[styles.circleDark1, {backgroundColor: COLORS.SPOTIFY_GREEN, shadowColor: COLORS.SPOTIFY_GREEN}]} />
            <View style={[styles.circleDark2, {backgroundColor: COLORS.SPOTIFY_GREEN, shadowColor: COLORS.SPOTIFY_GREEN}]} />
            <View style={[styles.circleDark3, {backgroundColor: COLORS.SPOTIFY_GREEN, shadowColor: COLORS.SPOTIFY_GREEN}]} />
            <View style={[styles.circleDark4, {backgroundColor: 'rgba(10, 10, 10, 1)', shadowColor: 'rgba(10, 10, 10, 1)'}]} />
            <View style={[styles.circleDark5, {backgroundColor: 'rgba(10, 10, 10, 1)', shadowColor: 'rgba(10, 10, 10, 1)'}]} />
            <View style={[styles.circleDark6, {backgroundColor: COLORS.SPOTIFY_PURPLE, shadowColor: COLORS.SPOTIFY_PURPLE}]} />
            <View style={[styles.circleDark7, {backgroundColor: COLORS.SPOTIFY_PURPLE, shadowColor: COLORS.SPOTIFY_PURPLE}]} />
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        position: 'absolute',
        width: '100%',
        height: '100%',
    },
    circleDark1: {
        ...circleStyle(200),
        top: 510,
        right: 150,
        opacity: 0.2,
    },
    circleDark2: {
        ...circleStyle(400),
        bottom: -200,
        right: -100,
        opacity: 0.3
    },
    circleDark3: {
        ...circleStyle(100),
        bottom: 280,
        right: 90,
        opacity: 0.1,
    },
    circleDark4: {
        ...circleStyle(200),
        top: 140,
        left: 30,
    },
    circleDark5: {
        ...circleStyle(150),
        top: 150,
        right: 40,
    },
    circleDark6: {
        ...circleStyle(50),
        top: 120,
        right: 150,
        opacity: 0.2
    },
    circleDark7: {
        ...circleStyle(250),
        top: -60,
        right: -60,
    },
});

function circleStyle(size) {
    return {
        position: 'absolute',
        width: size,
        height: size,
        borderRadius: size / 2,
        opacity: 0.4,
        shadowOffset: { width: 0, height: 0 },
        shadowOpacity: 0.8,
        shadowRadius: 6,
        elevation: 5,
    };
}

export default DarkBackgroundCircles;