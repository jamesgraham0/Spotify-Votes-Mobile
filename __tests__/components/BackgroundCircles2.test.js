import React from 'react';
import renderer from 'react-test-renderer';
import BackgroundCircles2 from '../../client/src/components/BackgroundCircles2';

describe('BackgroundCircles2', () => {
    it('should render the BackgroundCircles2 component', () => {
        const tree = renderer.create(<BackgroundCircles2 />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});