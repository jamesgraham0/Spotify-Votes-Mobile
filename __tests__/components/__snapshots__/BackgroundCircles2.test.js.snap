// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackgroundCircles2 should render the BackgroundCircles2 component 1`] = `
<View
  style={
    {
      "height": "100%",
      "position": "absolute",
      "width": "100%",
    }
  }
>
  <View
    style={
      [
        {
          "borderRadius": 100,
          "elevation": 5,
          "height": 200,
          "opacity": 0.2,
          "position": "absolute",
          "right": 150,
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 6,
          "top": 510,
          "width": 200,
        },
        {
          "backgroundColor": "#1DB954",
          "shadowColor": "#1DB954",
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "borderRadius": 200,
          "bottom": -200,
          "elevation": 5,
          "height": 400,
          "opacity": 0.3,
          "position": "absolute",
          "right": -100,
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 6,
          "width": 400,
        },
        {
          "backgroundColor": "#1DB954",
          "shadowColor": "#1DB954",
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "borderRadius": 50,
          "bottom": 280,
          "elevation": 5,
          "height": 100,
          "opacity": 0.1,
          "position": "absolute",
          "right": 90,
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 6,
          "width": 100,
        },
        {
          "backgroundColor": "#1DB954",
          "shadowColor": "#1DB954",
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "borderRadius": 100,
          "elevation": 5,
          "height": 200,
          "left": 30,
          "opacity": 0.4,
          "position": "absolute",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 6,
          "top": 140,
          "width": 200,
        },
        {
          "backgroundColor": "rgba(10, 10, 10, 1)",
          "shadowColor": "rgba(10, 10, 10, 1)",
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "borderRadius": 75,
          "elevation": 5,
          "height": 150,
          "opacity": 0.4,
          "position": "absolute",
          "right": 40,
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 6,
          "top": 150,
          "width": 150,
        },
        {
          "backgroundColor": "rgba(10, 10, 10, 1)",
          "shadowColor": "rgba(10, 10, 10, 1)",
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "borderRadius": 25,
          "elevation": 5,
          "height": 50,
          "opacity": 0.2,
          "position": "absolute",
          "right": 150,
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 6,
          "top": 120,
          "width": 50,
        },
        {
          "backgroundColor": "#B026FF",
          "shadowColor": "#B026FF",
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "borderRadius": 125,
          "elevation": 5,
          "height": 250,
          "opacity": 0.4,
          "position": "absolute",
          "right": -60,
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 6,
          "top": -60,
          "width": 250,
        },
        {
          "backgroundColor": "#B026FF",
          "shadowColor": "#B026FF",
        },
      ]
    }
  />
</View>
`;
